agent/http.go

```diff
@@ -414,15 +415,14 @@ func (s *HTTPServer) Index(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	// Give them something helpful if there's no UI so they at least know
-	// what this server is.
-	if !s.IsUIEnabled() {
+	if s.IsUIEnabled() {
+		// Redirect to the UI endpoint
+		http.Redirect(resp, req, "/ui/", http.StatusMovedPermanently) // 301
+	} else {
+		// Give them something helpful if there's no UI so they at least know
+		// what this server is.
```

Comment:
nit: general return in the conditional and continue unindented is cleaner go idiom than If/else. I see the comment was added but was there a reason to change the logic flow? Not a big deal.

Reply:
Good catch! I think I was trying to fix something, but I cannot remember what it was and I will remove that change.